liquibase-core/src/test/groovy/liquibase/change/core/CreateProcedureChangeTest.groovy

```diff
@@ -90,11 +95,91 @@ public class CreateProcedureChangeTest extends StandardChangeTest {
         new PostgresDatabase() | "all"                            | ""
     }
+
+    def dbmsIsNotConsideredOnCheckSumGeneration() {
+        when:
+        CreateProcedureChange change = new CreateProcedureChange()
+        change.setProcedureText(PROCEDURE_TEXT)
+        CheckSum procedureCheckSumWithoutDbms = change.generateCheckSum()
+        CreateProcedureChange change2 = new CreateProcedureChange()
+        change2.setProcedureText(PROCEDURE_TEXT)
+        change2.setDbms("postgresql")
+        CheckSum procedureCheckSumWithDbms = change2.generateCheckSum()
+
+        then:
+        procedureCheckSumWithoutDbms == procedureCheckSumWithDbms
+
+    }
+
+    def pathIsNotConsideredOnCheckSumGeneration() {
+        when:
+        String testScopeId = Scope.enter([
+                "resourceAccessor": new MockResourceAccessor([
+                        "test.sql": PROCEDURE_TEXT
+                ])
+        ])
+
+        CreateProcedureChange change = new CreateProcedureChange()
+        change.setProcedureText(PROCEDURE_TEXT)
+        CheckSum procedureCheckSumWithoutPath = change.generateCheckSum()
+        CreateProcedureChange change2 = new CreateProcedureChange()
+        change2.setProcedureText(PROCEDURE_TEXT)
+        change2.setPath("test.sql")
+        change2.setRelativeToChangelogFile(false)
+        //Below check sum generation should not take either path nor relativeToChangeLogFile properties into account
+        CheckSum procedureCheckSumWithPath = change2.generateCheckSum()
+        //TODO: Move this Scope.exit() call into a cleanUpSpec method
+        Scope.exit(testScopeId)
+
+        then:
+        procedureCheckSumWithoutPath == procedureCheckSumWithPath
+    }
+
+    def commentIsNotConsideredOnCheckSumGeneration() {
+        when:
+        CreateProcedureChange change = new CreateProcedureChange()
+        change.setProcedureText(PROCEDURE_TEXT)
+        CheckSum procedureCheckSumWithoutComments = change.generateCheckSum()
+        CreateProcedureChange change2 = new CreateProcedureChange()
+        change2.setProcedureText(PROCEDURE_TEXT)
+        change2.setComments("This is a test")
+        CheckSum procedureCheckSumWithComments = change2.generateCheckSum()
+
+        then:
+        procedureCheckSumWithoutComments == procedureCheckSumWithComments
+    }
+
+    def encodingIsNotConsideredOnCheckSumGeneration() {
+        when:
+        CreateProcedureChange change = new CreateProcedureChange()
+        change.setProcedureText(PROCEDURE_TEXT)
+        CheckSum procedureCheckSumWithoutEncoding = change.generateCheckSum()
+        CreateProcedureChange change2 = new CreateProcedureChange()
+        change2.setProcedureText(PROCEDURE_TEXT)
+        change2.setEncoding("UTF-8")
+        CheckSum procedureCheckSumWithEncoding = change2.generateCheckSum()
+
+        then:
+        procedureCheckSumWithoutEncoding == procedureCheckSumWithEncoding
+    }
+
+    def procedureTextUpdatedIsNotConsideredOnCheckSumGeneration() {
```

Comment:
It might be better to name this test to indicate that whitespace changes to procedure check will not impact the checksum. Non-whitespace changes should (and do) change the checksum.

Reply:
Sure, I can do that.

Reply:
Updating this test name with `procedureTextUpdatedWithWhitespacesAddedShouldNotComputeANewCheckSum`. Maybe too long, but I think it's pretty self-explainable.